### DISCUSSION TITLE
Does launch browser choice affect SEB stability?

### DISCUSSION THREAD

**Discussion by VirtualRusty** [SafeExamBrowser/seb-mac#265] [Q&A]
I work in an exam room with mixed BYOD and the Macs are the most likely to have issues with SEB, and we do most of our SEB restarts on Macs. I've been told that some browsers brands are preferred over others (Safari, Chrome, Firefox) but my question is - do these browsers have any affect on the SEB? Doesn't the launch browser just pass a link over to SEB which is a stand alone browser? Does SEB use WebKit independently of the launch browser or is there some other integration? Some background info would be helpful in understanding the environment.

**Comment by danschlet** [comment]
Can you be more specific about the issues you are having with SEB on Macs? We in contrary have far less support issues for SEB on Macs than on Windows.

Which assessment system are you using? With current SEB for macOS versions (at least 3.1.1)?

No, the browser you start an exam with doesn't have any influence on SEB stability. But the macOS version does, as SEB has to use the WebKit version of that macOS version. Also the system's WebKit version is NOT updated, when Apple updates Safari for that macOS version. You should demand that your students only use the latest three macOS versions, which are still getting security updates from Apple (currently macOS 11, 12 and 13).

Also you should make sure your assessment system uses the [SEB Javascript API and the modern WebView](https://safeexambrowser.org/developer/seb-config-key.html) in SEB (you can find that out by trying pinch-to-zoom on the trackpad, if it works it's the modern WebView).

**Reply by VirtualRusty** [reply]
I work as an invigilator in a 250 seat room and we have an average mix of about 65% Mac and 35% Win in student BYOD. Unfortunately the Macs require more SEB restarts than Win machines. I'm a Mac user so no bias, and its common knowledge with all the support people that Macs require more visits during the exam. I suspect they have issues with the PDFs in the exam module. We use Inspera. I might start to mentally tally whether its PDFs that are the issue or if its just general. It might be that base spec macs can't handle the graphics, but that's a wild guess.

We use 2.3.2 as that is that matches the current environment - I'm not involved in that part of the delivery, I'm just at the coalface.

We can't police what version of Mac OS students bring into the room. We have plenty of swap Win laptops if there's a failure. As a rule I try to keep the Macs going via an SEB restart as that only takes about 2 minutes.

The reason I asked about the browser is I was told by the 'team' that launch browser choice made a difference, and I was wary as I thought SEB was standalone and had nothing to do with the launch browser, which turned out to be true.

**Comment by danschlet** [comment]
I guess the combination of Inspera with their use of quite modern Web APIs and the rather old SEB 2.3.2 with the classic web browser engine causes those issues. The classic WebView for example crashes the whole SEB app if something goes wrong in rendering the website, while the modern WebView uses separate web rendering processes which don't crash the whole app if they need to terminate. Also the classic WebView isn't updated by Apple since a while.

But Inspera was doing a lot of testing of SEB 3.x and I think they will very soon switch to the current 3.x version. This will surely improve the stability on the Mac side.

**Reply by VirtualRusty** [reply]
That's useful info. I wondered why we used an older version. One of our many issues is students web search SEB and download the latest version that isn't compatible with Inspera, so we have to uninstall and reinstall. Keeps me busy doing this in the room in the first 10 mins of each exam.
